src/main/java/com/nha/abdm/wrapper/hip/hrp/link/userInitiated/LinkService.java

```diff
@@ -131,24 +155,51 @@ public void onConfirm(ConfirmResponse confirmResponse) {
     List<CareContext> careContexts = requestLogService.getSelectedCareContexts(linkRefNumber);
 
     OnConfirmPatient onConfirmPatient = null;
+    ResponseEntity<RequestStatusResponse> hipResponse=null;
+    try {
+      log.info("Requesting HIP for verify otp in discovery");
+      VerifyOTP verifyOTP=VerifyOTP.builder().authCode(confirmResponse.getConfirmation().getToken())
```

Comment:
I am not sure of the workflow here. In on-init we are asking hip to generate otp.. and in on-confirm we are asking hip to verify it? Aren't we putting a lot of overhead on hip? 

We can clearly save some trips and hops by saving OTP against link reference number..and verify in the wrapper itself instead of asking hip

Reply:
We are in the assumption that the otp is generated and validated at HIP itself, or else we can fetch the top in request/otp itself.